Audiobook Player/AppDelegate.swift

```diff
@@ -20,6 +20,14 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
         Fabric.with([Crashlytics.self])
+        let defaults:UserDefaults = UserDefaults.standard
+        
+        //If app has never been launched...
+        if defaults.bool(forKey: UserDefaultsConstants.isFirstLaunch) {
```

Comment:
maybe we can move this `if` to the bottom of the function before `return true` and do something like:
```
.
.
.
UserDefaults.standard.set(nil, forKey: "sleep_timer")

//here comes the new code
if defaults.bool(forKey: UserDefaultsConstants.completedFirstLaunch) {
  return true
}

//Set smartRewindEnabled and completedFirstLaunch to true
defaults.set(true, forKey: UserDefaultsConstants.smartRewindEnabled)
defaults.set(true, forKey: UserDefaultsConstants.completedFirstLaunch)

return true
}//end of function
```
it's not set in stone to call the constant `completedFirstLaunch`, I used that name in efforts to make clear what I mean 🤔. What do you think ?

Reply:
Good catch! I originally had the method reversed and then decided to "clean it up" grammatically, and ended up with bad logic. I like using "completedFirstLaunch" as the constant for clarity, but I personally like having a !completedFirstLaunch check and have all initial setup logic contained in there. It makes it a bit easier to read at a glance since it's all chunked in one spot and you can't miss the `return true` and be confused when reading it. This way we can always add any fresh "first run" logic and always know where it's all contained. I'm thinking about changing it to:

```
//If app has never been launched...
        if !defaults.bool(forKey: UserDefaultsConstants.completedFirstLaunch) {
            //Set smartRewindEnabled and isFirstLaunch to false
            defaults.set(true, forKey: UserDefaultsConstants.smartRewindEnabled)
            defaults.set(false, forKey: UserDefaultsConstants.completedFirstLaunch)
            
            //Ensure user defaults are saved immediately
            defaults.synchronize()
        }
```

What do you think?

Reply:
makes sense what you say about `return true` 👌, let's go with your suggestion, but don't forget in that block of code to remove `defaults.synchronize` and to set to `true` `completedFirstLaunch` 👍